tests/test_time.py

```diff
@@ -119,8 +122,9 @@ def test_shuffle(self):
         model.step()
         for output in self.expected_output[:2]:
             assert output in model.log[:2]
-        for output in self.expected_output[2:]:
-            assert output in model.log[2:]
+        for output in self.expected_output[3:]:
```

Comment:
I suppose the existing code isn't clear about saying that it checks that the shuffling happens within stages, and not across stages.